src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Markup/XamlGridLengthSerializer.cs

```diff
@@ -187,21 +187,20 @@ static internal void FromString(
             out double       value,
             out GridUnitType unit)
         {
-            string goodString = s.Trim().ToLowerInvariant();
+            ReadOnlySpan<char> valueSpan = s.Trim().AsSpan();
 
             value = 0.0;
             unit = GridUnitType.Pixel;
 
             int i;
-            int strLen = goodString.Length;
             int strLenUnit = 0;
             double unitFactor = 1.0;
 
             //  this is where we would handle trailing whitespace on the input string.
             //  peel [unit] off the end of the string
             i = 0;
 
-            if (goodString == UnitStrings[i])
+            if (valueSpan.Equals(UnitStrings[i], StringComparison.OrdinalIgnoreCase))
```

Comment:
I used Ordinal because the existing comparison uses Ordinal (== uses Ordinal) and I believe the "ignore case" part of OrdinalIgnoreCase is the same as InvariantCultureIgnoreCase so I believe the code is correct.

Reply:
@ThomasGoulet73 Could you see https://stackoverflow.com/a/72766 ? It says that the OrdinalIgnoreCase  is not same as InvariantCultureIgnoreCase .

Reply:
Ordinal[IgnoreCase] is certainly not the same as Invariant[IgnoreCase]. You can see it e.g. on ae and æ. However, I don't think that is a concern for the involved strings here (auto, px, *).

Reply:
I think using Ordinal[IgnoreCase] will be better than Invariant[IgnoreCase] in this case.

Reply:
Same.

Reply:
> Ordinal[IgnoreCase] is certainly not the same as Invariant[IgnoreCase]. You can see it e.g. on ae and æ. However, I don't think that is a concern for the involved strings here (auto, px, *).

Sorry if I wasn't clear, what I meant is better explained [here](https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#:~:text=Comparisons%20made%20using%20StringComparison.OrdinalIgnoreCase%20are%20behaviorally%20the%20composition%20of%20two%20calls%3A%20calling%20ToUpperInvariant%20on%20both%20string%20arguments%2C%20and%20doing%20a%20comparison%20using%20StringComparison.Ordinal.).

Basically, it says that `value.ToLowerInvariant().Equals(value2.ToLowerInvariant(), StringComparison.Ordinal)` is supposed to be the same as `value.Equals(value2, StringComparison.OrdinalIgnoreCase)` which is what I meant.